src/profiler.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AuthenticationConfig, Common, Logger, Service, ServiceConfig} from '../third_party/types/common-types';
+
+import {Config, ConfigImpl} from './config';
+import {CpuProfiler} from './profilers/cpu-profiler';
+import {HeapProfiler} from './profilers/heap-profiler';
+
+const common: Common = require('@google-cloud/common');
+
+const pjson = require('../../../package.json');
+const path = require('path');
+
+// TODO: finish implementing Profiler.
+// TODO: add stop() method to stop profiling.
+export class Profiler {
```

Comment:
If you say "export" here, does it mean it's visible to other files in this directory or is it visible to the clients of the package? I wouldn't expect the latter for now.

Reply:
I'm not totally certain if this makes the class visible to clients of the package, but I do know it's necessary to make this class visible to other files in the same directory.

Reply:
It would be good to understand how we can limit the visibility to clients. That's probably something very common to do in Node.js.

Reply:
Exporting here makes it visible to anyone able to `require`/`import` this file. This would typically be code internal to this module, but this is JavaScript at the end of the day and it is possible for anyone to access anything anyway. There is no good way to limit visibility to external users.

The way it is written, this is not _supposed_ to be a public interface. I think that is good enough.